SpeedLimiterMsg path
SpeedLimiterMsg shadow
SpeedLimiterMsg obstacle
SpeedLimiterMsg external